packages/engine/paima-runtime/src/runtime-loops.ts

```diff
@@ -57,94 +56,123 @@ async function runPresync(
   startBlockHeight: number,
   stopBlockHeight: number | null
 ): Promise<void> {
-  const maximumPresyncBlockheight = stopBlockHeight
-    ? Math.min(startBlockHeight, stopBlockHeight)
-    : startBlockHeight;
   let presyncBlockHeight = await getPresyncStartBlockheight(
     gameStateMachine,
     funnelFactory.getExtensions(),
-    startBlockHeight,
-    maximumPresyncBlockheight
+    startBlockHeight
   );
 
-  if (presyncBlockHeight > maximumPresyncBlockheight) {
-    // doLog(
-    //   `[paima-runtime] Skipping presync (presync block height: ${presyncBlockHeight}, maximum presync block height: ${startBlockHeight})`
-    // );
-    return;
+  if (!ENV.CARP_URL) {
+    delete presyncBlockHeight[Network.CARDANO];
   }
```

Comment:
Could this not cause a subtle issue where if you specify a Cardano CDE but the CARP_URL is missing that it just silently doesn't track anything? I feel if you're using task that requires Carp and you didn't specify the CARP_URL it should be an error

Reply:
yeah, makes sense, changed it for an error